### ISSUE TITLE
Contacted David Stafford

### ISSUE THREAD

**Issue by LiHe93** [ParadoxSoftware/jet3d-devel#5]
I managed to contact David Stafford regarding relicensing Genesis3D/Jet3D under a more liberal license and have received his blessings.
I can send you screenshots of the convo.
I think it's about time to revive this! 😀

**Comment by paradoxnj** [comment]
Please provide the screenshots.  Not sure why you didn't in the first place.  :/  Licensing was never the issue as all our changes are open source anyway.  The issue was time, number of developers and the old codebase being too outdated.  Are you offering your help?  If not, I am working on it in my spare time as I have other obligations which are more important.

**Comment by LiHe93** [comment]
![Jet3d 2](https://user-images.githubusercontent.com/2000845/73943487-7f6b0c80-48f9-11ea-8852-965a15c43837.png)
![Jet3d license](https://user-images.githubusercontent.com/2000845/73943492-8134d000-48f9-11ea-88ac-7191ed736add.png)
Ha sorry
I am not a very good coder, and i actually prefer working in C rather than C++ which is why i was drawn to Genesis3D in the first place.
But i'd love to help in any way that i can! Especially if the license would be liberal now

**Comment by paradoxnj** [comment]
The license was liberal to begin with.  The only requirement was that you open source your changes which is the same as the LGPL and display the logo on any games written with it.  :/  The license never stopped us from developing it...people's time did.  People would show up, stay for a few months, realize the engine code was terrible, then left me to clean up their mess.

Writing a 3D engine is hard.  Modernizing a really old engine (22 years old now) is even harder.  If you are not a good coder, there is not much you can really do.  I appreciate the sentiment though.

This is going to require a lot of time.  As I am the lone coder here with my spare time as my availability, this will not move fast at all.  For example, to make this 64 bit, I need to remove all inline assembly.  Because Jet3D was written in 1998, the engine is full of assembly (check out ExtBox.cpp in the code for example).  I have to translate that assembly code into C/C++ which is a lot of work as I am not a really good assembly coder.  So in lieu of 64 bit, I have been working on an OpenGL driver which uses modern rendering techniques.  This way I can at least improve the performance and efficiency of the engine.  From there, I will start moving rendering calls to the engine itself thus removing the driver code which is terrible.  I am not in a rush to convert to classes.

**Comment by LiHe93** [comment]
LGPL-like licenses prevent you from using closed source components or compile to consoles which require an NDA.
About the x86 assembly, maybe take a shortcut and use something like an x86 vm?
Something like: https://github.com/tboox/vm86/blob/master/README.md

About the 32/64-bit thing i'm not sure, but if i recall correctly then 64-bit software can use 32-bit libraries. So maybe DLL-ing the 64-bit incompatible code would yeild some results?

**Comment by paradoxnj** [comment]
You can use closed source components as long as you don't statically link the library.  If you use it in a DLL as it is meant to be used, you don't need to open source your application.

You would need to spend money to develop for consoles anyway.  Jet3D was not written for consoles. It was written for PC.  If you are going to adapt to console, you may as well write a new engine as it would be less effort than adapting Jet3D to them.  Remember Jet3D uses BSP rendering which is very outdated by today's standards.  If you remove the BSP from Jet3D, it is no longer Jet3D.  Just write a new engine at that point.

Using DLLs is possible, but that is not clean or efficient.  I'll take a look at the VM.  Like I said...no rush for 64 Bit.  OpenGL is more important at the moment.

**Comment by LiHe93** [comment]
OpenGL is a good bet. 
And i really like BSP based tools, and with the advent of new Retro-shooters i wouldn't be so quick to try and replace it.

When it comes to consoles it's actually relatively easy to start developing for them nowadays. Microsoft gives the Xbox dev kit pretty much free to indies, nintendo and sony are pretty lax too.
Genesis3D was used on the Gamecube,
But I don't know how different it is from Jet3D.

**Comment by paradoxnj** [comment]
DirectX7 is intertwined into the engine itself thus making it hard for console use.  I ma having issues completely removing it.

The subtractive BSP (Unreal 1 Style) is really cool and makes the whole engine.  The object system is very bad as all objects are DLLs or Plugins to the engine.   That is not very expandable.  I'd like to remove that and make it more like G3D in that respect where you add objects to your game code.  The world file format is very bad as well.  It is just a bunch of objects in no particular order saving themselves.  Those are the next priorities.

**Comment by LiHe93** [comment]
I see. It might be a bit much for me to work on jet3d then.
You mind then if i fork G3D using your OpenGL driver under a different license?

**Comment by paradoxnj** [comment]
Yes...I do mind.  The license is liberal enough.  There is zero reason to change it.  It kind of sounds to me like you want unrestricted access to code without giving credit.  That is all that the license does not allow.

**Comment by LiHe93** [comment]
I apologize it seems that way, that is not my intention at all.
Thank you for your help
